source/common/router/router.cc

```diff
@@ -576,10 +576,21 @@ void Filter::handleNon5xxResponseHeaders(const Http::HeaderMap& headers, bool en
   }
 }
 
+void Filter::onUpstream100ContinueHeaders(Http::HeaderMapPtr&& headers) {
+  ENVOY_STREAM_LOG(debug, "upstream 100 continue", *callbacks_);
+
+  downstream_response_started_ = true;
+  // We will double count response codes for 100-Continue.
+  upstream_request_->upstream_host_->outlierDetector().putHttpResponseCode(100);
+  // Don't send retries after 100-Continue has been sent on.
```

Comment:
technically, I think we could still retry by resending expect: continue again, and eating the next 100 continue, but I agree this is overcomplicated and not necessary. I might expand this comment a bit though about that potential and why we are disabling retry, with an optional TODO to implement retry if you feel that would ever happen.

Reply:
I'd argue that's not the right behavior.  Technically someone could have different backends with different behaviors and the next would not accept.  I mean it's not the end of the world but it is the start of a response.  Will definitely comment.